File: 코딩테스트/백준유형/순열조합/2798_조합.cpp
#include <iostream>
#include <vector>
using namespace std;

int N, M;
// int arr[100];

int maxx;
int sum;


int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> N >> M;
    vector<int>arr(N);
    for (int i=0; i<N; i++)
    {
        cin >> arr[i];
    }

    // 조합

    for (int i=0; i<N; i++)
    {
        for (int j=i+1; j<N; j++)
        {
            for (int k=j+1; k<N; k++)
            {
                sum = arr[i] + arr[j] + arr[k];
                if (sum > maxx && sum <= M) maxx = sum;
            }
        }
    }
    cout << maxx << "\n";
    
    return 0;
}

File: 코딩테스트/백준유형/순열조합/2798_조합.swift
import Foundation

var N: Int = 0
var M: Int = 0
var maxx: Int = 0
var sum: Int = 0

// 표준 입력을 위해 readLine 사용
if let input = readLine() {
    let inputs = input.split(separator: " ").map { Int(String($0))! }
    N = inputs[0]
    M = inputs[1]
}

// 배열 초기화
var arr: [Int] = []

if let arrayInput = readLine() {
    arr = arrayInput.split(separator: " ").map { Int(String($0))! }
}

// 조합
for i in 0..<N {
    for j in i+1..<N {
        for k in j+1..<N {
            sum = arr[i] + arr[j] + arr[k]
            if sum > maxx && sum <= M {
                maxx = sum
            }
        }
    }
}

print(maxx)
